pkg/synthetictests/kubelet.go

```diff
@@ -462,6 +474,21 @@ func buildTestsFailIfRegexMatch(testName string, matchRE, dontMatchRE *regexp.Re
 			continue
 		}
 
+		// Skip if this ErrImagePull problem is part of a negative test (i.e., if any of these
+		// patterns match the event string, it is an expected ErrImagePull failure).
+		found := false
+		for i := 0; i < len(expectedErrPatterns); i++ {
+			if expectedErrPatterns[i].MatchString(estr) {
+				found = true
+				break
+			}
+		}
+		if found {
+			// ErrImagePull was found but also one of the expected error patters was found so don't
+			// count this as an ErrImagePull failure.
+			continue
```

Comment:
This is very readable so no need  to change anything but go supports labels with goto/continue/break:

```go
Outer:
for _, i := range outer {
  for _, j := range inner {
    if i == j {
      break Outer
    }
  }
}
```